### PR TITLE
feat: BIND integration, allow Argentinian users to make deposits in ARS

### PR DESCRIPTION
**Pull request by lsena-bc**
## Description
Allow users from Argentina to make deposits in the wallet to buy crypto with ARS currency. Also we split the code of the bank details to allow developers to add another new currency more easily.

## Testing Steps
1. Create an Argentinian (AR) user.
2. In backoffice:
  - Request document (only one is enough)
  - Update the user document using this one 20312528046
  - Update the user address
  - Approved KYC
  - Verifiy tier 2
3. Login inside the wallet.
4. Select the ARS currency.
5. Make a deposit in ARS.
6. And you will see the bank transfer details for Argentinians.

Screen Shot 2022-05-23 at 15 33 04

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Deposit/WireInstructions/template.success.tsx [hunk 1]
```diff
@@ -77,16 +69,12 @@ const TabsContainer = styled.div`
 const Success: React.FC<Props> = (props) => {
   const [transferType, setTransferType] = useState(TransferType.DOMESTIC)
 
-  const formatIbanAddress = (): string => {
-    const EVERY_FOUR_CHARS = /(.{4})(?!$)/g
-    return props.account.address.replace(EVERY_FOUR_CHARS, `$1 `)
+  // TODO: Improve currency layouts logic
```
Comment (julianromero-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Deposit/WireInstructions/template.success.tsx:72]: Are you planning to resolve this TODO in next PR? if not you should create a jira task with a more specific description in what to do

File: packages/blockchain-wallet-v4-frontend/src/data/coins/selectors/coins/ars.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import { FormattedMessage } from 'react-intl'
+
+// retrieves introduction text for coin on its transaction page
+export const getTransactionPageHeaderText = () => (
+  <FormattedMessage
+    id='coins.usd.intro'
```
[the remark this answers is not in the document]
Comment (lsena-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/coins/selectors/coins/ars.tsx:7]: On it!

File: packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Deposit/WireInstructions/template.success.tsx (hunk 1 above)
Comment (lsena-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Deposit/WireInstructions/template.success.tsx:72]: I'm not sure if I solved it with good practices, if you have a feedback about how I resolved it please feel free to tell me any suggestion to fix it before we send this functionality to production. If not I will remove the TODO comment 😝

**Review by julianromero-bc — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/utils/selectors.ts [hunk 3]
```diff
@@ -22,6 +23,14 @@ export const getCoinsWithBalanceOrMethod = (state: RootState) => {
   const erc20sR = selectors.core.data.eth.getErc20AccountTokenBalances(state)
   const recentSwapTxs = selectors.custodial.getRecentSwapTxs(state).getOrElse([] as SwapOrderType[])
   const custodials = selectors.core.data.coins.getCustodialCoins()
+  // TODO: Move this code to another place to clean this file
+  const userData = selectors.modules.profile.getUserData(state).getOrElse({} as UserDataType)
+  const userCountryCode = userData?.address?.country || 'default'
+  const countryCurrencies = {
+    AR: ['ARS'],
+    default: ['USD', 'EUR', 'GBP']
+  }
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/utils/selectors.ts:32]: i dont like the idea of hardcoding this list in yet another spot. please find a way to use `WalletFiatEnum` and `userCountryCode` to build the `fiatCurrencies`

**Review by schnogz — CHANGES_REQUESTED** [review]

File: packages/blockchain-wallet-v4-frontend/src/data/components/utils/selectors.ts (hunk 3 above)
Comment (lsena-bc) [reply packages/blockchain-wallet-v4-frontend/src/data/components/utils/selectors.ts:39]: @schnogz Done!

**Review by schnogz — APPROVED** [review]

**Pull request merged**